contracts/staking/validatorShare/ValidatorShare.sol

```diff
@@ -236,6 +232,60 @@ contract ValidatorShare is IValidatorShare, ERC20NonTransferable, OwnableLockabl
         }
     }
 
+    /**
+        New shares exit API
+     */
+
+    function sellVoucher_new(uint256 claimAmount, uint256 maximumSharesToBurn) public {
+        // first get how much staked in total and compare to target unstake amount
+        (uint256 totalStaked, uint256 rate) = getTotalStake(msg.sender);
+        require(totalStaked > 0 && totalStaked >= claimAmount, "Too much requested");
+
+        // convert requested amount back to shares
+        uint256 precision = _getRatePrecision();
+        uint256 shares = claimAmount.mul(precision).div(rate);
+        require(shares <= maximumSharesToBurn, "too much slippage");
+
+        _withdrawAndTransferReward(msg.sender);
+
+        _burn(msg.sender, shares);
+        stakeManager.updateValidatorState(validatorId, -int256(claimAmount));
+
+        uint256 _withdrawPoolShare = claimAmount.mul(precision).div(withdrawExchangeRate());
+        withdrawPool = withdrawPool.add(claimAmount);
+        withdrawShares = withdrawShares.add(_withdrawPoolShare);
+
+        uint256 unbondNonce = unbondNonces[msg.sender].add(1);
```

Comment:
Should we update after we set it by`unbonds_new[msg.sender][unbondNonce] = unbond` ? So it can start with `0`? or it doesn't matter I guess.